File: core/engines/fullseason.py
# core/engines/fullseason.py
from core.model_engine import create_model_engine


def create_fullseason_engine(DEFAULT_STATS, DEFAULT_W, DEFAULT_W_VAR, BAYES_HYPER, ENGINE_CONFIG):
    return create_model_engine(
        DEFAULT_STATS,
        DEFAULT_W,
        DEFAULT_W_VAR,
        BAYES_HYPER,
        engine_config=ENGINE_CONFIG,
        enable_h2h=True,
        enable_team_hca=True,   # team-specific HCA — applied only when caller passes team_hca dict (run_daily wires it in playoffs only)
        bayes={
            "spread": {
                "s_diff_cap": None,
                "abs_line_cap": 18,
                "abs_line_cap_inclusive": False,
            },
            "totals": {"enabled": False},
        },
    )

File: core-js/engines/fullseason.mjs
import { createModelEngine } from "../model_engine.mjs";

export function createFullseasonEngine({ DEFAULT_STATS, DEFAULT_W, DEFAULT_W_VAR, BAYES_HYPER }) {
  return createModelEngine({
    DEFAULT_STATS,
    DEFAULT_W,
    DEFAULT_W_VAR,
    BAYES_HYPER,
    enableH2H: true,
    bayes: {
      spread: { sDiffCap: null, useSDiff: false },
      totals: { enabled: false },
    },
  });
}
